review/searchTab.js

```diff
@@ -0,0 +1,108 @@
+import { CLASS_LIST, URL, KEYCODE } from '../util/data';
+import { li, makeAutoCompleteItem, makeRecommendItem, ul } from '../util/htmlTemplate';
+import { autoCompleteParser } from '../util/parser';
+import { getData, _, debounce } from '../util/util';
+
+function SearchTab({ data, selector }) {
+  this.recommendData = data;
+  this.searchTab = selector.searchTab;
+  this.searchInput = selector.searchInput;
+  this.searchTabTitle = selector.searchTabTitle;
+  this.currentIdx = -1;
+  this.orginInput;
+  this.autoCompleteData;
+  this.timer;
+}
+
+SearchTab.prototype = {
+  constructor: SearchTab,
+  init() {
+    this.registerEvent();
+    this.renderSearchTab();
+  },
+  registerEvent() {
+    this.searchInput.addEventListener('input', this.handleInput.bind(this));
+    this.searchInput.addEventListener('keydown', this.handleKeydown.bind(this));
+  },
+  handleInput({ target: { value } }) {
+    this.timer = debounce(this.render.bind(this, value), 1000, this.timer);
+  },
+  handleKeydown({ keyCode }) {
+    if (keyCode === KEYCODE.UP) this.moveUpList();
+    else if (keyCode === KEYCODE.DOWN) this.moveDownList();
+  },
+  getRecommendHTML() {
+    const { SEARCH_TAB_LIST } = CLASS_LIST;
+    let firstList = '';
+    let secondList = '';
+    this.recommendData.forEach((data, idx) => {
+      if (idx < this.recommendData.length / 2) firstList += makeRecommendItem(idx + 1, data);
+      else secondList += makeRecommendItem(idx + 1, data);
+    });
+    const recommendHTML =
+      ul({ value: firstList, classes: [SEARCH_TAB_LIST] }) + ul({ value: secondList, classes: [SEARCH_TAB_LIST] });
+    return recommendHTML;
```

Comment:
그리고 `firstList`와 `secondList`를 만드는 일을 하나의 메서드로 만들고 각각 호출하는 방식으로 변경해보면 어떨까요?
```javascript
_makeRecommandedItemsHTML(start, end) {
  // start부터 end index에 해당하는 html template string 반환
}
```

```suggestion
const { length } = this.recommendData;
const half = Math.ceil(length / 2);
return this._makeRecommandedItemsHTML(0, half) + this._makeRecommandedItemsHTML(half, length);
```